shell/platform/android/jni/platform_view_android_jni.h

```diff
@@ -0,0 +1,122 @@
+#define FLUTTER_SHELL_PLATFORM_ANDROID_PLATFORM_VIEW_ANDROID_JNI_H_
+
+#include <any>
+
+#include "flutter/fml/macros.h"
+#include "flutter/fml/mapping.h"
+#include "flutter/lib/ui/window/platform_message.h"
+
+#include "third_party/skia/include/core/SkMatrix.h"
+
+namespace flutter {
+
+//------------------------------------------------------------------------------
+/// Allows to call Java code running in the JVM from any thread. However, most
+/// methods can only be called from the platform thread as that is where the
+/// Java code runs.
+///
+/// This interface must not depend on the Android toolchain directly, so it can
+/// be used in unit tests compiled with the host toolchain.
+///
+class PlatformViewAndroidJNI {
+ public:
+  virtual ~PlatformViewAndroidJNI();
+
+  //----------------------------------------------------------------------------
+  /// @brief      Sends a platform message. The message may be empty.
+  ///
+  virtual void FlutterViewHandlePlatformMessage(
+      fml::RefPtr<flutter::PlatformMessage> message,
+      int responseId) = 0;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Responds to a platform message. The data may be a `nullptr`.
+  ///
+  virtual void FlutterViewHandlePlatformMessageResponse(
+      int responseId,
+      std::unique_ptr<fml::Mapping> data) = 0;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Sends semantics tree updates.
+  ///
+  /// @note       Must be called from the platform thread.
+  ///
+  virtual void FlutterViewUpdateSemantics(std::vector<uint8_t> buffer,
+                                          std::vector<std::string> strings) = 0;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Sends new custom accessibility events.
+  ///
+  /// @note       Must be called from the platform thread.
+  ///
+  virtual void FlutterViewUpdateCustomAccessibilityActions(
+      std::vector<uint8_t> actions_buffer,
+      std::vector<std::string> strings) = 0;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Indicates that FlutterView should start painting pixels.
+  ///
+  /// @note       Must be called from the platform thread.
+  ///
+  virtual void FlutterViewOnFirstFrame() = 0;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Indicates that a hot restart is about to happen.
+  ///
+  virtual void FlutterViewOnPreEngineRestart() = 0;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Attach the SurfaceTexture to the OpenGL ES context that is
+  ///             current on the calling thread.
+  ///
+  /// @note       `surface_texture` must be `fml::jni::JavaObjectWeakGlobalRef`
+  ///
+  virtual void SurfaceTextureAttachToGLContext(std::any surface_texture,
```

Comment:
`std::any` still leads to needing casts in Android code and the loss of readability of the type of the surface texture and its consequences to IDE completions. While there are very real uses of `std::any`, in this case, it is little more than a glorified raw pointer.

I would much prefer something like the following in a common header.
```
#if OS_ANDROID
using JavaWeakGlobalRef = fml::jni::JavaObjectWeakGlobalRef;
#else
using JavaWeakGlobalRef = std::nullptr_t;
#endif
```

This avoid casts, the type is clear and IDE completions all just work.

Reply:
Done in https://github.com/flutter/engine/pull/18903/commits/8d9d48adbaef79c0936de582803a7fa72193f023